test/extended/testdata/router/router-h2spec.yaml

```diff
@@ -0,0 +1,165 @@
+apiVersion: v1
```

Comment:
I guess we should talk about nginx... if it's possible to re-do with haproxy, benefits include:

1. We can test the same haproxy image we're deploying for ingress
2. No concerns about using an image we're not shipping
3. ???

Downsides include:

1. You already did it in nginx

Reply:
I think it makes sense to go for haproxy - I just wanted the basic infrastructure up as a PR so we could discuss it.

Reply:
I actually considered an axis of `haproxy, nginx` which is why I named these as such. Perhaps that's overkill. Thoughts?

The test would effectively then be:

    for backed in "nginx", "haproxy", do...

Reply:
Switched to just haproxy. Squashed commits.